From a0e357899523d68da044bd0e2b27c724e7974b80 Mon Sep 17 00:00:00 2001
From: Torge Matthies <openglfreak@googlemail.com>
Date: Mon, 3 Jun 2024 11:05:38 +0200
Subject: [PATCH 29/53] winegstreamer: Fix race between
 wg_parser_stream_disable and GST_EVENT_FLUSH_STOP.

CW-Bug-Id: #21644
---
 dlls/winegstreamer/wg_parser.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/dlls/winegstreamer/wg_parser.c b/dlls/winegstreamer/wg_parser.c
index c8dc49cdbd6..bb4dcda0556 100644
--- a/dlls/winegstreamer/wg_parser.c
+++ b/dlls/winegstreamer/wg_parser.c
@@ -996,8 +996,7 @@ static gboolean sink_event_cb(GstPad *pad, GstObject *parent, GstEvent *event)
             pthread_mutex_lock(&parser->mutex);
 
             stream->eos = false;
-            if (stream->enabled)
-                stream->flushing = false;
+            stream->flushing = false;
 
             pthread_mutex_unlock(&parser->mutex);
             break;
-- 
2.47.1

